### PR TITLE
Fixed #31210 -- Documented how to reproduce HttpRequest.is_ajax()

### PR DESCRIPTION
**Pull request by adamchainz**

### CODE REVIEW

File: docs/releases/3.1.txt [hunk 1]
```diff
@@ -475,12 +475,16 @@ Miscellaneous
...
-  <https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API>`_. Depending on
-  your use case, you can either write your own AJAX detection method, or use
-  the new :meth:`.HttpRequest.accepts` method if your code depends on the
-  client ``Accept`` HTTP header.
+  jQuery-specific way of signifying AJAX calls, while current usage tends to
+  use the JavaScript `Fetch API
+  <https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API>`_.
+  Depending on your use case, you can either write your own AJAX detection
+  method, or use the new :meth:`.HttpRequest.accepts` method if your code
+  depends on the client ``Accept`` HTTP header.
```
Comment (felixxm) [inline docs/releases/3.1.txt]: Please revert all unnecessary changes in the existing description.

File: docs/releases/3.1.txt [hunk 2]
```diff
@@ -475,12 +475,16 @@ Miscellaneous
...
+
+  If you are writing your own AJAX detection method, the behavior of
+  ``HttpRequest.is_ajax()`` can be reproduced exactly with
```
Comment (felixxm) [inline docs/releases/3.1.txt]: Maybe ```the behavior of deprecated ``request.is_ajax()`` method can be ... ```

File: docs/releases/3.1.txt (hunk 1 above)
Comment (adamchainz) [reply docs/releases/3.1.txt]: I fixed the phrase "while the current usage tends to use" which doesn't really make sense,  the rest is diff noise from re-wrapping

File: docs/releases/3.1.txt (hunk 2 above)
Comment (adamchainz) [reply docs/releases/3.1.txt]: I think we don't need to repeat the word "deprecated" within the release note that it's deprecated

Comment (felixxm) [reply docs/releases/3.1.txt]: And what about ``request`` instead of ``HttpRequest``?

File: docs/releases/3.1.txt (hunk 1 above)
Comment (felixxm) [reply docs/releases/3.1.txt]: I missed that you removed `the`.

File: docs/releases/3.1.txt (hunk 2 above)
Comment (adamchainz) [reply docs/releases/3.1.txt]: Done

**Comment by felixxm** [comment]
@adamchainz Thanks :+1:

**Comment by collinanderson** [comment]
Thanks!
